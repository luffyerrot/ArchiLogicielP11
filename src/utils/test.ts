import { ComponentMountingOptions, mount, shallowMount, VueWrapper } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import { expect, it } from "vitest";
import { createVuetify } from "vuetify";
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

export function factory(originalComponent: any, shallow = false, pinia = false, options?: ComponentMountingOptions<any, unknown>): VueWrapper<unknown, any> {
    if (pinia) {
        setActivePinia(createPinia());
    }
    if (shallow) {
        return shallowMount(originalComponent, options);
    } else {
        const vuetify = createVuetify({
            components,
            directives,
        });
        const defaultOptions = {
            global: {
                plugins: [vuetify]
            },
            ...options
        };
        return mount(originalComponent, defaultOptions);
    }
};

export function renderCorrectly(wrapper: VueWrapper<unknown, any>) {
    it('Bon affichage de la page', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })
}