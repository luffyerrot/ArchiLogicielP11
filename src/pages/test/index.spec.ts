import { factory, renderCorrectly } from "@/utils/test";
import { describe } from "vitest";
import Index from "../index.vue";

describe('Page home', () => {
    const wrapper = factory(Index, true);
    renderCorrectly(wrapper);
})