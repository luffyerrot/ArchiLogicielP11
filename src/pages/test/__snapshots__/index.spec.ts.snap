// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page home > Bon affichage de la page 1`] = `
"<h1 data-v-d1d3d0d7="" class="message ma-4">Bienvenue sur le service de recherche MedHead</h1>
<div data-v-d1d3d0d7="">
  <p data-v-d1d3d0d7="" class="message ma-4" style="color: red;">Erreur lors de la récupération des groupes de spécialités</p>
</div>
<v-container-stub data-v-d1d3d0d7="" fluid="false" tag="div"></v-container-stub>"
`;
